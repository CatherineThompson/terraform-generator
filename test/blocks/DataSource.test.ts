import { DataSource } from '../..';
import { tfGenerator11, tfGenerator12, arg4 } from '.';

test('DataSource 0.11', () => {
  const dataSource = new DataSource(tfGenerator11, 'type', 'name', arg4(tfGenerator11));
  expect(dataSource.toTerraform()).toMatchSnapshot();
  expect(dataSource.getIdentifier().toTerraform()).toMatchSnapshot();
  expect(dataSource.getAttribute('attr').toTerraform()).toMatchSnapshot();
});

test('DataSource 0.12', () => {
  const dataSource = new DataSource(tfGenerator12, 'type', 'name', arg4(tfGenerator12));
  expect(dataSource.toTerraform()).toMatchSnapshot();
  expect(dataSource.getIdentifier().toTerraform()).toMatchSnapshot();
  expect(dataSource.getAttribute('attr').toTerraform()).toMatchSnapshot();
});
