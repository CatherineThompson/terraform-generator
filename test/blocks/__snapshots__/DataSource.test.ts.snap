// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSource 0.11 1`] = `
"data \\"type\\" \\"name\\"{
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg14 = {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
}
}

"
`;

exports[`DataSource 0.11 2`] = `"data.type.name"`;

exports[`DataSource 0.11 3`] = `"data.type.name.attr"`;

exports[`DataSource 0.12 1`] = `
"data \\"type\\" \\"name\\"{
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg14 = {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
}
}

"
`;

exports[`DataSource 0.12 2`] = `"data.type.name"`;

exports[`DataSource 0.12 3`] = `"data.type.name.attr"`;
